### PR TITLE
feat: support decoding of RADON error codes and messages

### LINKED ISSUE #30: Ensure compatibility of Rust and solidity RAD errors

### LINKED ISSUE #44: Support `RadonError::Timeout`
The `RadonError::Timeout` error case should be supported by the `Witnet.Result` library.

### PR DESCRIPTION
**Pull request by aesedepece**
Fix #30
Fix #44

### CODE REVIEW

File: contracts/Witnet.sol [hunk 1]
```diff
@@ -153,13 +247,53 @@ library Witnet {
...
+    } else if (_stageIndex == 2) {
+      return "tally";
+    } else {
+      return "unknown";
+    }
+  }
+
+  /**
+   * @notice Convert a `uint8` into a 1, 2 or 3 characters long `string` representing its decimal value
+   * @param _u A `uint8` value
```
Comment (mariocao) [inline contracts/Witnet.sol]:
```suggestion
   * @param _u A `uint64` value
```

File: contracts/Witnet.sol [hunk 2]
```diff
@@ -70,13 +101,76 @@ library Witnet {
...
+      errorMessage = abi.encodePacked("Overflow at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.DivisionByZero) {
+      errorMessage = abi.encodePacked("Division by zero at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.RuntimeError) {
+      errorMessage = abi.encodePacked("Unspecified runtime error at operator code ", utoa(error[4]), " found at call #", utoa(error[3]), " in script #", utoa(error[2]), " from ", stageName(error[1]), " stage");
+    } else if (errorCode == ErrorCodes.InsufficientConsensusError) {
+      errorMessage = abi.encodePacked("Insufficient consensus. Required: ", utoa(error[1]), ". Achieved: ", utoa(error[2]));
+    } else {
+      errorMessage = abi.encodePacked("Unknown error (", utoa(error[0]), ")");
+    }
```
Comment (mariocao) [inline contracts/Witnet.sol]: Knowing that the formatting rule is a real pain...
`in case of more than 3 arguments, drop each into its own line.`

**Comment by aesedepece** [comment]
Formatter should pass now :wink:

**Comment by aesedepece** [comment]
This is finally ready for review :)

**Comment by aesedepece** [comment]
Error codes now match the ones from https://github.com/witnet/witnet-rust/blob/5da3f0ae9773f3e822e9b01751c417a9bc164de1/data_structures/src/radon_error.rs

**Pull request closed without merging**